azure/scope/identity.go

```diff
@@ -165,17 +157,10 @@ func (p *AzureCredentialsProvider) GetTokenCredential(ctx context.Context, resou
 		}
 		cred, authErr = NewWorkloadIdentityCredential(azwiCredOptions)
 
-	case infrav1.ServicePrincipal, infrav1.ServicePrincipalCertificate, infrav1.UserAssignedMSI:
-		if err := createAzureIdentityWithBindings(ctx, p.Identity, resourceManagerEndpoint, activeDirectoryEndpoint, clusterMeta, p.Client); err != nil {
-			return nil, err
-		}
-
-		options := azidentity.ManagedIdentityCredentialOptions{
-			ID: azidentity.ClientID(p.Identity.Spec.ClientID),
-		}
-		cred, authErr = azidentity.NewManagedIdentityCredential(&options)
+	case infrav1.ServicePrincipalCertificate, infrav1.UserAssignedMSI:
+		return nil, errors.Errorf("identity type %s not supported", p.Identity.Spec.Type)
```

Comment:
This would be breaking, is there no way to use the SDK to get the token via those methods?

Reply:
I reimplemented these two methods using the `azidentity` API, but I'm not sure where to fetch the certificate data in the case of `infrav1.ServicePrincipalCertificate`.